src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -160,7 +168,7 @@ public void testFromXContent() throws IOException {
     public void testToQuery() throws IOException {
         testQuery = createTestQueryBuilder();
         QueryParseContext context = createContext();
-        context.setMapUnmappedFieldAsString(true);
+        context.setAllowUnmappedFields(true);
```

Comment:
I need to correct myself, we need to set the default to allow unmapped fields, otherwise the String mapper wins over the date mapper we introduced here. Don't understand why, but setAllowUnmappedFields(true) worked with the other tests so far as well.